src/clj_puppetdb/http.clj

```diff
@@ -24,47 +24,86 @@
   `puppetlabs.ssl-utils.core/pems->ssl-context` function."
   [:ssl-cert :ssl-key :ssl-ca-cert])
 
+(defn- make-client-common
+  [^String host opts]
+  (let [opts (assoc opts :as :stream)
+        info {:host host}
+        info (if-let [ssl-context (:ssl-context opts)]
+               (assoc info :ssl-context ssl-context)
+               info)]
+    (fn
+      ; This arity-overloaded variant of the function is the one called by the `GET` function.
+      ; The `this` parameter refers under normal circumstances to this very function and it is
+      ; needed to be able to call another arity-overloaded variant of the function from the
+      ; function body. When the `this` parameter refers to a different function then it
+      ; effectively overrides the other arity-overloaded variants of the function.
+      ([this path params]
```

Comment:
Since (as explained above) we need the [this path params] arity overloaded variant to make the overriding possible the best we can do is to define yet another arity overloaded variant like so:
```clojure
    (fn stadarad_client
      ([path params]
        (standard_client standard_client path params))
      ...
```
we would have to add a similar variant to the vcr client as well:
```clojure
    (fn vcr_client
      ([path params]
        (vcr_client vcr_client path params))
      ...
```
With this in place we could call the client from the ```GET``` function simply as:
```clojure
  (client path params)
```
But I consider this a cosmetic change which unnecessarily adds another function call.